File: 프로그래머스/Level 2/가장 큰 정사각형 찾기.swift
// https://school.programmers.co.kr/learn/courses/30/lessons/12905?language=swift
import Foundation

func solution(_ board:[[Int]]) -> Int {
    var answer:Int = board[0][0]
    var board = board
    
    for i in 1..<board.count {
        for j in 1..<board[i].count {
            if board[i][j] == 1 {
                board[i][j] = 1 + min(board[i-1][j-1],board[i-1][j], board[i][j-1])
                answer = max(answer, board[i][j])
            }
        }
    }

    return answer*answer
}

File: 프로그래머스/Level 2/가장 큰 정사각형 찾기.cpp
// https://school.programmers.co.kr/learn/courses/30/lessons/12905
#include <iostream>
#include <vector>

using namespace std;

int findMin(vector<int> v) {
    int ret = v[0];
    
    for (int i = 1; i < v.size(); ++i) {
        if (ret > v[i]) ret = v[i];
    }
    
    return ret;
}

int solution(vector<vector<int>> board) {
    int answer = board[0][0];
    
    for (int i = 1; i < board.size(); ++i) {
        for (int j = 1; j < board[0].size(); ++j) {
            if (board[i][j] == 1) {
                int tmp = findMin({board[i-1][j], board[i][j-1], board[i-1][j-1]});
                board[i][j] = 1 + tmp;
                answer = answer > board[i][j] ? answer : board[i][j];
            }
        }
    }
    
    // for (int i = 0; i < board.size(); ++i) {
    //     for (int j = 0; j < board[0].size(); ++j) {
    //         cout << board[i][j] << ' ';
    //     }
    //     cout << '\n';
    // }

    return answer * answer;
}